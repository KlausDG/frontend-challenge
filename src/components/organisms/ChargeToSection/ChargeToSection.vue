<script setup lang="ts">
import { Card, Stack } from '@/components/atoms'
import { IconText } from '@/components/molecules'
import type { Address } from '@/models'

defineProps<{
  data: Address
}>()
</script>

<template>
  <section>
    <Card title="Charge to" :subtitle="data.name">
      <Stack direction="column">
        <Stack>
          <Stack direction="column" :spacing="4" class="content__column">
            <IconText
              icon="file-landscape-alt"
              :text="
                data.document
                  ? `${data.document.type}: ${data.document.value}`
                  : 'CNPJ: 00.000.000/0000-00'
              "
            />
            <IconText icon="map-marker" :text="data.address" />
            <IconText icon="user" :text="data.contact.name" />
          </Stack>
          <Stack direction="column" :spacing="4">
            <IconText icon="envelope" :text="data.contact.email" />
            <IconText icon="phone" :text="data.contact.phone" />
            <IconText icon="print" :text="data.contact.fax" />
          </Stack>
        </Stack>
      </Stack>
    </Card>
  </section>
</template>

<style lang="scss" scoped>
.subtitle__container {
  align-items: center;
}

.content__column {
  width: 50%;
}
@media (min-width: 1024px) {
}
</style>
