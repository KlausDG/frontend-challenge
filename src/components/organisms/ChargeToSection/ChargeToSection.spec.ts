import { mount } from '@vue/test-utils'
import { describe, it, expect } from 'vitest'
import ChargeToSection from './ChargeToSection.vue'

const CardMock = {
  name: 'Card',
  props: ['title', 'subtitle'],
  template: '<div class="mock-card"><slot /></div>',
}

const StackMock = {
  name: 'Stack',
  props: ['direction', 'spacing'],
  template: '<div class="mock-stack"><slot /></div>',
}

const IconTextMock = {
  name: 'IconText',
  props: ['icon', 'text'],
  template: '<div class="mock-icon-text">{{ icon }}:{{ text }}</div>',
}

describe('ChargeToSection.vue', () => {
  const addressData = {
    name: 'Company Ltd',
    address: '456 Business Rd',
    label: 'teste',
    code: '123',
    contact: {
      name: 'Alice Smith',
      email: 'alice@example.com',
      phone: '123456789',
      fax: '987654321',
    },
    document: {
      type: 'CNPJ',
      value: '12.345.678/0001-90',
    },
  }

  const wrapper = mount(ChargeToSection, {
    props: { data: addressData },
    global: {
      components: {
        Card: CardMock,
        Stack: StackMock,
        IconText: IconTextMock,
      },
    },
  })

  it('renders Card with correct title and subtitle', () => {
    const card = wrapper.findComponent(CardMock)
    expect(card.exists()).toBe(true)
    expect(card.props('title')).toBe('Charge to')
    expect(card.props('subtitle')).toBe(addressData.name)
  })

  it('renders top-level Stack', () => {
    const stack = wrapper.findComponent(StackMock)
    expect(stack.exists()).toBe(true)
  })

  it('renders all IconText components with correct props', () => {
    const iconTexts = wrapper.findAllComponents(IconTextMock)
    expect(iconTexts).toHaveLength(6)

    expect(iconTexts[0].props()).toEqual({
      icon: 'file-landscape-alt',
      text: `${addressData.document.type}: ${addressData.document.value}`,
    })
    expect(iconTexts[1].props()).toEqual({ icon: 'map-marker', text: addressData.address })
    expect(iconTexts[2].props()).toEqual({ icon: 'user', text: addressData.contact.name })
    expect(iconTexts[3].props()).toEqual({ icon: 'envelope', text: addressData.contact.email })
    expect(iconTexts[4].props()).toEqual({ icon: 'phone', text: addressData.contact.phone })
    expect(iconTexts[5].props()).toEqual({ icon: 'print', text: addressData.contact.fax })
  })

  it('renders content__column class correctly', () => {
    const columnStack = wrapper.find('.content__column')
    expect(columnStack.exists()).toBe(true)
  })

  it('matches the component snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
