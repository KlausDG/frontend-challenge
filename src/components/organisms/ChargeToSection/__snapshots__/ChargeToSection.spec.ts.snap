// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ChargeToSection.vue > matches the component snapshot 1`] = `
"<section data-v-026a4ebc="">
  <div data-v-69f42c1a="" data-v-9bb7b3b9="" data-v-026a4ebc="" class="stack__container card__container" style="flex-direction: column; gap: 16px;">
    <div data-v-69f42c1a="" data-v-9bb7b3b9="" class="stack__container" style="flex-direction: column; gap: 4px;">
      <p data-v-9bb7b3b9="" class="typography _caption_d0cbd2 _text_d0cbd2">Charge to</p>
      <p data-v-9bb7b3b9="" class="typography _semibold_d0cbd2 _text_d0cbd2">Company Ltd</p>
    </div>
    <div data-v-69f42c1a="" data-v-026a4ebc="" class="stack__container" style="flex-direction: column; gap: 0px;">
      <div data-v-69f42c1a="" data-v-026a4ebc="" class="stack__container" style="flex-direction: row; gap: 0px;">
        <div data-v-69f42c1a="" data-v-026a4ebc="" class="stack__container content__column" style="flex-direction: column; gap: 4px;">
          <div data-v-70c01fa4="" data-v-026a4ebc="" class="icon-text__container">
            <unicon data-v-70c01fa4="" name="file-landscape-alt"></unicon>
            <p data-v-70c01fa4="" class="typography _caption_d0cbd2 _text_d0cbd2">CNPJ: 12.345.678/0001-90</p>
          </div>
          <div data-v-70c01fa4="" data-v-026a4ebc="" class="icon-text__container">
            <unicon data-v-70c01fa4="" name="map-marker"></unicon>
            <p data-v-70c01fa4="" class="typography _caption_d0cbd2 _text_d0cbd2">456 Business Rd</p>
          </div>
          <div data-v-70c01fa4="" data-v-026a4ebc="" class="icon-text__container">
            <unicon data-v-70c01fa4="" name="user"></unicon>
            <p data-v-70c01fa4="" class="typography _caption_d0cbd2 _text_d0cbd2">Alice Smith</p>
          </div>
        </div>
        <div data-v-69f42c1a="" data-v-026a4ebc="" class="stack__container" style="flex-direction: column; gap: 4px;">
          <div data-v-70c01fa4="" data-v-026a4ebc="" class="icon-text__container">
            <unicon data-v-70c01fa4="" name="envelope"></unicon>
            <p data-v-70c01fa4="" class="typography _caption_d0cbd2 _text_d0cbd2">alice@example.com</p>
          </div>
          <div data-v-70c01fa4="" data-v-026a4ebc="" class="icon-text__container">
            <unicon data-v-70c01fa4="" name="phone"></unicon>
            <p data-v-70c01fa4="" class="typography _caption_d0cbd2 _text_d0cbd2">123456789</p>
          </div>
          <div data-v-70c01fa4="" data-v-026a4ebc="" class="icon-text__container">
            <unicon data-v-70c01fa4="" name="print"></unicon>
            <p data-v-70c01fa4="" class="typography _caption_d0cbd2 _text_d0cbd2">987654321</p>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>"
`;
